pkg/buildermgr/pkgwatcher.go

```diff
@@ -301,3 +565,16 @@ func setInitialBuildStatus(fissionClient *crd.FissionClient, pkg *fv1.Package) (
 	// TODO: use UpdateStatus to update status
 	return fissionClient.CoreV1().Packages(pkg.Namespace).Update(pkg)
 }
+
+// setFinalBuildStatus sets initial build status to a package if it is empty.
+// This normally occurs when the user applies package YAML files that have no status field
+// through kubectl.
+func setFinalBuildStatus(fissionClient *crd.FissionClient, pkg *fv1.Package) (*fv1.Package, error) {
```

Comment:
I think there are two things - 
- build: A package built from source code
- builder: A pod used to build a package from source

The build should have status - from Pending-> Running-> Succeeded/Failed 

But the builder itself is a transient entity - which gets created when a package is created or updated and then disappears after the building is done. So the status of the builder should not go in the status of the build at all.

What would be ideal is - as the builder is created running and deleted - the events on the package should be updated.